src/go/MONIT/alert.go

```diff
@@ -0,0 +1,658 @@
+		UpdateTimestamp  time.Time `json:"updateTimestamp"`
+	} `json:"annotations"`
+	StartsAt time.Time `json:"startsAt"`
+	EndsAt   time.Time `json:"endsAt"`
+}
+
+//AlertManager GET API acceptable JSON Data struct for SSB data
+type ssbData struct {
+	Data []amSSBJSON
+}
+
+//AlertManager API acceptable JSON Data for Grafana Alerts
+type amGrafanaJSON struct {
+	Labels struct {
+		Alertname string `json:"alertname"`
+		Severity  string `json:"severity"`
+		Service   string `json:"service"`
+		Tag       string `json:"tag"`
+	} `json:"labels"`
+	Annotations struct {
+		Description string `json:"description"`
+		Summary     string `json:"summary"`
+	} `json:"annotations"`
+	StartsAt time.Time `json:"startsAt"`
+	EndsAt   time.Time `json:"endsAt"`
+}
+
+//AlertManager GET API acceptable JSON Data struct for Grafana Alerts
+type grafanaData struct {
+	Data []amGrafanaJSON
+}
+
+//Alert CLI tool data struct (Tabular)
+type alertData struct {
+	Name     string
+	Service  string
+	Tag      string
+	Severity string
+	StartsAt time.Time
+	EndsAt   time.Time
+}
+
+//Array of alerts for alert CLI Tool (Tabular)
+var allAlertData []alertData
+
+//Alert CLI tool data struct (JSON)
+type alertDataJSON struct {
+	Name     string
+	Service  string
+	Tag      string
+	Severity string
+	Starts   string
+	Ends     string
+	Duration string
+}
+
+//Array of alerts for alert CLI Tool (JSON)
+var allAlertDataJSON []alertData
+
+//function for get request on /api/v1/alerts alertmanager endpoint for fetching alerts.
+func get(service string, data interface{}) {
+
+	var apiurl string
+	//GET API for fetching only GGUS alerts.
+	if service == "" {
+		apiurl = cmsmonURL + "/api/v1/alerts?active=true&silenced=false&inhibited=false&unprocessed=false"
+	} else {
+		apiurl = cmsmonURL + "/api/v1/alerts?active=true&silenced=false&inhibited=false&unprocessed=false&filter=service=" + service
+	}
+
+	req, err := http.NewRequest("GET", apiurl, nil)
+	req.Header.Add("Accept-Encoding", "identity")
+	req.Header.Add("Accept", "application/json")
+
+	client := &http.Client{}
+
+	if verbose > 1 {
+		dump, err := httputil.DumpRequestOut(req, true)
+		if err == nil {
+			log.Println("Request: ", string(dump))
+		}
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	byteValue, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		log.Printf("Unable to read %s JSON Data from AlertManager GET API, error: %v\n", service, err)
+		return
+	}
+
+	err = json.Unmarshal(byteValue, &data)
+	if err != nil {
+		log.Printf("Unable to parse %s JSON Data from AlertManager GET API, error: %v\n", service, err)
+		return
+	}
+
+	if verbose > 1 {
+		dump, err := httputil.DumpResponse(resp, true)
+		if err == nil {
+			log.Println("Response: ", string(dump))
+		}
+	}
+
+}
+
+//function for merging all alerts from various services at one place
+func mergeData(amGGUSData ggusData, amSSBJSON ssbData, amGrafanaData grafanaData) {
```

Comment:
Please convert it to use variadic input, e..g
```
// here data is array of amData structs, but you don't restrict how many you can pass
fund mergeData(arrData ...amData)
```
and loop over a given array of amData structs. The code repeats single for loop for every given struct which you should generalize. Learn about variadic input, e.g. https://gobyexample.com/variadic-functions

Reply:
Valentin,
With the generalized version of amData structs. mergeData now doesn't need multiple arguments. Thanks for the information though.